crates/chia-bls/src/secret_key.rs

```diff
@@ -242,8 +242,15 @@ impl SecretKey {
     #[classattr]
     pub const PRIVATE_KEY_SIZE: usize = 32;
 
-    pub fn sign_g2(&self, msg: &[u8]) -> crate::Signature {
-        crate::sign(self, msg)
+    pub fn sign(&self, msg: &[u8], final_pk: Option<PublicKey>) -> crate::Signature {
```

Comment:
I understand the python interface has this optional public key, but do we ever actually use it? If not, it would be cleaner to just not have it